### PR TITLE
Revised all ACSL except for what does this program do, Recursion, and LISP

### PR DESCRIPTION
**Pull request by nishikarp**
I added some more explanation about binary trees in the in-fix. post-fix, and pre-fix section, but not sure if it makes sense so take a look at that at part?

### CODE REVIEW

File: resources/acsl/assembly.md [hunk 1]
```diff
@@ -149,7 +149,7 @@ Look at the following table below to understand how the program changes over tim
 | MULT | 8 | 1 | 4 |
 | STORE | 8 | 1 | 8 |
 | LOAD | 1 | 1 | 8 |
-| SUB | 0 | 1 | 8 |
+| SUB | `$0$` | 1 | 8 |
 
 Now that ACC is equal to '$0$', the branch instruction BE after SUB now applies,
```
Comment (Stinggyray) [inline resources/acsl/assembly.md:154]: Lol that's backticks

File: resources/acsl/assembly.md [hunk 2]
```diff
@@ -149,7 +149,7 @@ Look at the following table below to understand how the program changes over tim
 | MULT | 8 | 1 | 4 |
 | STORE | 8 | 1 | 8 |
 | LOAD | 1 | 1 | 8 |
-| SUB | 0 | 1 | 8 |
+| SUB | `$0$` | 1 | 8 |
```
Comment (Stinggyray) [inline resources/acsl/assembly.md:152]: If you're going to make this LaTeX make every other number LaTeX too

Comment (Stinggyray) [reply resources/acsl/assembly.md:152]: Not sure why you want to use LaTeX though. Just stick to regular numbers in my opinion

(So convert everything back to just numbers maybe?)

File: resources/acsl/booleanalgebra.md [hunk 3]
```diff
@@ -37,32 +37,29 @@ Here's an example:
 
 | X | Y | X and Y |
 | --- | --- | --- |
-| 0 | 0 | 0 |
-| 1 | 0 | 0 |
-| 0 | 1 | 0 |
-| 1 | 1 | 1 |
+| `$0$` | `$0$` | `$0$` |
+| `$1$` | `$0$` | `$0$` |
```
Comment (Stinggyray) [inline resources/acsl/booleanalgebra.md:41]: Same deal, not sure why you LaTeXed this

File: resources/acsl/booleanalgebra.md [hunk 4]
```diff
@@ -216,12 +213,12 @@ First off, let's simplify this expression:
...
 Now, we can set this final reduced expression to 1, since the problem asks us to find ordered pairs to make the expression true. Then, using a truth table, the solutions can be easily found:
 
-| X | Y | `$X + \overline{Y}$` |
-| --- | --- | --- |
-| 0 | 0 | 1 |
-| 1 | 0 | 1 |
-| 0 | 1 | 0 |
-| 1 | 1 | 1 |
+| X | Y | `$\overline{Y}$`|`$X + \overline{Y}$` |
+| --- | --- | ---|--- |
```
Comment (Stinggyray) [inline resources/acsl/booleanalgebra.md:217]: What's with the wonky table separator? Lol

File: resources/acsl/datastructures.md [hunk 5]
```diff
@@ -48,21 +48,22 @@ different, as it is written as `X = POP()`. What this does is that it removes th
 element in the stack and stores it in variable X. So, if there was a stack with its
 top element being "E", then writing `H = POP()` would store "E" in H.
 
-In the case that the POP operation is called on an empty stack, then the variable is given the special value NIL, which means "nonexistent".
+In the case that the POP operation is called on an empty stack, then the variable is given the special value NIL(Not In List), which means "nonexistent".
```
Comment (Stinggyray) [inline resources/acsl/datastructures.md:51]: Add a space before parenthesis please

File: resources/acsl/datastructures.md [hunk 6]
```diff
@@ -48,21 +48,22 @@ different, as it is written as `X = POP()`. What this does is that it removes th
...
 
 ## Queues
 
-Queues, on the other hand, are often used to handle requests. It's similar to waiting in line (or a *queue* in Britain); the first people in line are processed first. Hence, queues follow a "first in, first out" (FIFO) order.
+Queues, on the other hand, are often used to handle requests. It's similar to waiting in line (or a *queue*); the first people in line are processed first. Hence, queues follow a "first in, first out" (FIFO) order.
 
-The PUSH operation works the same as it would with stacks. However, with POP, instead of removing the top (most recently added) element, queues have their bottom element removed.
+The PUSH operation works the same as it would with stacks(the element is added to the top of the data structure). However, with POP, instead of removing the top (most recently added) element, queues have their bottom element removed.
 
+A tip to handle both queue and stack related problems is to keep a list of all the elements, with the pushed elements being added to the top of the list. In a queue, cross of the bottom element, and in a stack cross of the top element.
```
Comment (Stinggyray) [inline resources/acsl/datastructures.md:59]: Cross off... nishikar stop using the wrong off lol

File: resources/acsl/datastructures.md [hunk 7]
```diff
@@ -74,7 +75,7 @@ Each node can have a total of two children. The left child must be less than or
 
 ### Inserting Nodes (Binary Search)
 
-Inserting a node requires knowing its position relative to existing nodes. Here's an example with the word *AMERICAN*:
+Inserting a node requires knowing its position relative to existing nodes Starting at the root node, compare the node you are inserting to another node to determine whether the node will go to the right or left of the node you are comparing it to. If there is no exisiting node, insert the node there, or repeat this process with the node that is there until.  Here's an example with the word *AMERICAN*:
```
Comment (Stinggyray) [inline resources/acsl/datastructures.md:78]: exisiting

Some sentences here don't make grammatical sense. Read them carefully?

File: resources/acsl/datastructures.md [hunk 8]
```diff
@@ -87,17 +88,17 @@ Inserting a node requires knowing its position relative to existing nodes. Here'
...
+Note that the order you insert the nodes does matter, so if you were to switch out the positions of the letters in *AMERICAN* to become something like *MCEAANIR*, the tree would look very different.
 
 ### Deleting Nodes (Binary Search)
 
-In the case that a node needs to be deleted, this generally involves some minor shifting. Less, equal, and greater relationships between nodes still have to be considered for proper placement.
+In the case that a node needs to be deleted, there are three cases you must consider: the removed node has `$0$` children, `$1$` child, or `$2$` children. When there are no children, you can simply remove the node. When there is one child, move the child node and its children up to replace the parent node. When there are two children, move the left child and its children up to replace the parent node, and add on the right child and all of its children to the left child or its children, using the same comparison process as described above. Removing nodes generally involves some minor shifting because "less than or equal to" and "greater than" relationships between nodes still have to be considered for proper placement.
```
Comment (Stinggyray) [inline resources/acsl/datastructures.md:95]: Don't just use LaTeX everywhere for numbers lol

File: resources/acsl/datastructures.md [hunk 9]
```diff
@@ -164,7 +174,7 @@ See the table below to understand the differences between the two in more detail
 
 # Path Lengths
 
-Before moving further into priority queues, let's talk briefly about path lengths. In earlier sections, I addressed "layers"/"levels" of the tree; these can also be referred to as "depths". The root has a depth of 0; the next layer has a depth of "1", and this continues to increment for subsequent, deeper layers.
+Before moving further into priority queues, let's talk briefly about path lengths. In earlier sections, I addressed "layers"/"levels" of the tree; these can also be referred to as "depths". The root node has a depth of `$0$`; the next layer has a depth of "1", and this continues to increment for subsequent, deeper layers.
```
Comment (Stinggyray) [inline resources/acsl/datastructures.md:177]: Please treat your LaTeX addiction nishikar

File: resources/acsl/digitalelectronics.md [hunk 10]
```diff
@@ -9,7 +9,7 @@
 
 Digital electronics is a way to visualize boolean algebra using drawn-out circuits (also
 known as logic gates). Be sure to make sure you understand all of the boolean operations;
-check out the [Boolean Algebra](/resources/acsl/booleanalgebra) page if needed.
+check out the [Boolean Algebra](/resources/acsl/booleanalgebra) page if needed. If you understood that page really well, this page will be a piece of cake for you.
```
Comment (Stinggyray) [inline resources/acsl/digitalelectronics.md:12]: Something more along the lines of "If you understand boolean algebra, you understand digital electronics" would be better

File: resources/acsl/digitalelectronics.md [hunk 11]
```diff
@@ -77,14 +77,14 @@ be made to get the solution.
...
-| 0 | 1 | 0 | 0 | 0 | 0 |
-| 0 | 0 | 1 | 0 | 1 | 0 |
-| 1 | 1 | 0 | 1 | 1 | 1 |
-| 1 | 0 | 1 | 1 | 0 | 0 |
-| 0 | 1 | 1 | 0 | 1 | 0 |
-| 1 | 1 | 1 | 1 | 0 | 0 |
+| `$0$` | `$0$` | `$0$` | `$1$`  | `$1$`  | `$0$` |
+| `$1$` | `$0$` | `$0$` | `$1$`  | `$1$`  | `$1$`  |
+| `$0$` | `$1$`  | `$0$` | `$0$` | `$0$` | `$0$` |
+| `$0$` | `$0$` | `$1$`  | `$0$` | `$1$`  | `$0$` |
```
Comment (Stinggyray) [inline resources/acsl/digitalelectronics.md:83]: Why

File: resources/acsl/graphtheory.md [hunk 12]
```diff
@@ -124,17 +125,16 @@ To start off, our template matrix would look like so:
 
 Note that the letters are simply written as a reference; they are not part of the actual matrix.
 
-Then, we fill out our matrix with 0s and 1s, with 1 standing for if an edge between two specific vertices, from
-the vertex in the column to the vertex in the row, exists. So, for example, in the top right empty box, this would
-be filled out with a 1 because there is indeed an edge that goes from A to C.
+Then, we fill out our matrix with `$0$`s and `$1$`s, with `$1$` signifying that an edge connects the vertice in its column and its row. So, for example, in the top right empty box, this would
```
Comment (Stinggyray) [inline resources/acsl/graphtheory.md:128]: Same thing

File: resources/acsl/graphtheory.md [hunk 13]
```diff
@@ -124,17 +125,16 @@ To start off, our template matrix would look like so:
...
 
 So, the rest of the matrix will be filled out as follows:
 
 | | A | B | C |
 | --- | --- | --- | --- |
-| A | 1 | 0 | 1 |
-| B | 0 | 1 | 1 |
-| C | 1 | 0 | 0 |
+| A | `$1$` | `$0$` | 1 |
+| B | `$0$` | `$1$` | `$1$` |
```
Comment (Stinggyray) [inline resources/acsl/graphtheory.md:136]: Reeeee

File: resources/acsl/notation.md [hunk 14]
```diff
@@ -145,6 +145,11 @@ operations in the same row, and then move up to the next row up. In this next ro
...
 is continued until you get to the uppermost (or final) operation.
 
+Another way to think about converting from a tree to an expression is as follows:
+For in-fix, starting at the parent node, visit the left child node, then the parent node, and finally the right child node. If any of the child nodes have children, apply the same process to those nodes. For example, starting at the equals sign, visit the left X node, and since it has no children write it down. Do the same for the parent equals sign node. Then visit the right node. But this node has a left child so visit its left child before writing down this new node. Using the same logic, the multiplies node has a left child as well, so visist that. The A had no children so write that down, and then write down the multiply sign. Finally, visit the right node of the multiplies and since it has no children write down B. Now you have finally finished evaluating the left child of the minus sign, so now you can write down the minus node, and proceed to visit the right child of that node. Continue down this process, and get the equation `$X = (AB - C/D)\uparrow E$`.
```
Comment (Stinggyray) [inline resources/acsl/notation.md:149]: Wall of text. Please add breaks or organization

File: resources/acsl/numbersystems.md [hunk 15]
```diff
@@ -39,7 +39,7 @@ number; so for example, 10 is denoted as `A`, 11 is denoted as `B`, and so on.
 
 With these ranges, notice that the upper bound is one less than the base; so, with binary,
 which is base 2, the upper bound is `$2 - 1 = 0$`. Also notice that all of them have a
-lower bound of 0.
+lower bound of `$0$`.
```
Comment (Stinggyray) [inline resources/acsl/numbersystems.md:42]: pls

File: resources/acsl/numbersystems.md [hunk 16]
```diff
@@ -126,7 +127,9 @@ would be reduced to `$1111011_2$`.
 
 The same idea applies for if you want to convert from binary to hex and vice versa except that
 you work with groups of 4 bits instead of 3. Also, you will have to work with the letters A-E,
-which again, represent 10-15.
+which again, represent 10-15. 
+
+When converting between hex to octal or vice versa, you can use this trick to go to binary first and then to the desired base. This trick can be very helpful as the numbers you deal with are much smaller, leaves you much less prone to computing errors, and could potentially save time.
```
Comment (Stinggyray) [inline resources/acsl/numbersystems.md:132]: The last sentence has some grammar issues

File: resources/acsl/regex.md [hunk 17]
```diff
@@ -70,7 +70,7 @@ are a few things to keep in mind:
 2. If *a* and *b* are both regex, then so are the following:
     * `$ab$` - This is just *a* followed by *b*; it does not mean *and* like it would in boolean algebra. The proper term for this would be **concatenation**.
     * `$a\cup b$` or `$a\,|\,b$` - This stands for *a or b*. The proper term for this is *union*. (In almost every programming language or sensible implementation of regex, the pipe symbol `|` is used.)
-    * `$a*$` - This is *a* repeated 0 or more times. This is known as *closure*, or the *Kleene star* (named after Stephen Cole Kleene, the computer scientist who invented regular expressions and helped establish recursion theory).
+    * `$a*$` - This is *a* repeated `$0$`  or more times. This is known as *closure*, or the *Kleene star* (named after Stephen Cole Kleene, the computer scientist who invented regular expressions and helped establish recursion theory).
```
Comment (Stinggyray) [inline resources/acsl/regex.md:73]: Stop with the latex pls

File: resources/acsl/regex.md [hunk 18]
```diff
@@ -121,8 +121,8 @@ Also, we'll go over the actual syntax in programming languages further down
 
 | Symbol | Meaning | Example | Matches | Doesn't Match |
 | --- | --- | --- | --- | --- |
-| `/*/` | This matches the preceding token **0 or more** times. | `/ba*/` | `b` `ba`, `baa`, `baaaaaa` | `bbaa`, `aaa` |
-| `/?/` | This matches the preceding token **0 or 1** times (basically, makes the previous token *optional*). | `/colou?r/` | `color`, `colour` | `coloer` |
+| `/*/` | This matches the preceding token **`$0$` or more** times. | `/ba*/` | `b` `ba`, `baa`, `baaaaaa` | `bbaa`, `aaa` |
+| `/?/` | This matches the preceding token **`$0$`  or 1** times (basically, makes the previous token *optional*). | `/colou?r/` | `color`, `colour` | `coloer` |
```
Comment (Stinggyray) [inline resources/acsl/regex.md:125]: Reeeeee

**Review by Stinggyray — CHANGES_REQUESTED** [review]
oh boy

File: resources/acsl/datastructures.md (hunk 6 above)
Comment (nishikarp) [reply resources/acsl/datastructures.md]: It was one time lol

**Review by Stinggyray — APPROVED** [review]

**Pull request merged**